manifests/params.pp

```diff
@@ -13,25 +13,37 @@
         $service_file_location   = '/usr/lib/systemd/system/jira.service'
         $service_file_template   = 'jira/jira.service.erb'
         $service_lockfile        = '/var/lock/subsys/jira'
+        $service_provider        = 'systemd'
       } elsif $::operatingsystemmajrelease == '6' or $::operatingsystem == 'Amazon'{
         $json_packages           = [ 'rubygem-json', 'ruby-json' ]
         $service_file_location   = '/etc/init.d/jira'
         $service_file_template   = 'jira/jira.initscript.erb'
         $service_lockfile        = '/var/lock/subsys/jira'
+        $service_provider        = 'init'
       } else {
         fail("\"${module_name}\" provides no service parameters
             for \"${::osfamily}\" - \"${::operatingsystemmajrelease}\"")
       }
     } /Debian/: {
+      if $::operatingsystemmajrelease == '8' {
```

Comment:
Hmm, I think you misread @igalic's comment in that one. The regex approach actually enables us to *not* rely on future parser.

It's still no silver bullet, though, because you either lock yourself to one version

    if $::operatingsystemmajrelease =~ /^8/

or just a limited subset

    if $::operatingsystemmajrelease =~ /^[8-9]/

or you make something weird

    if $::operatingsystemmajrelease =~ /^[8-9]\.|^[1-9][0-9]/

and it is all sad because you actually wanted an integer comparison in the first place.

Perhaps it's the most sane approach to

 1. make sure the release is a string, by converting if necessary and
 2. actually use `versioncmp`

I like `versioncmp` because while it's overkill for Debian and friends, the `operatingsystemmajrelease` fact receives fun values such as `14.04` on Ubuntu for example. `versioncmp` is robust in such scenarios.